src/dotnet/commands/dotnet-compile-native/Program.cs

```diff
@@ -2,11 +2,13 @@
 using System.IO;
 using Microsoft.DotNet.Cli.Utils;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.DotNet.Tools.Compiler.Native
 {
     public class CompileNativeCommand
     {
+        private const string McgInteropAssemblySuffix = ".mcginterop.dll";
```

Comment:
This needs to be `.McgInterop` . It will not work on Unix otherwise because of Unix is case sensitive.